### PR TITLE
Document how to inject Configuration

### PR DESCRIPTION
**Pull request by cchantep**

### CODE REVIEW

File: documentation/manual/working/commonGuide/configuration/ConfigFile.md [hunk 1]
```diff
@@ -19,6 +19,21 @@ At runtime, the default `application.conf` is loaded from the classpath. System
 
 These system properties specify a replacement for `application.conf`, not an addition. If you still want to use some values from the `application.conf` file then you can include the `application.conf` in your other `.conf` file by writing `include "application"` at the top of that file. After you've included the `application.conf`'s settings in your new `.conf` file you can then specify any settings that you want override.
 
+## Using from your controller
+
+The configuration can be available in your controller (or your component), to use the default settings or your custom one, thanks to Dependency Injection (in [[Scala|ScalaDependencyInjection]] or in [[Java|JavaDependencyInjection]]).
+
+```scala
```
Comment (marcospereira) [inline documentation/manual/working/commonGuide/configuration/ConfigFile.md]: Please extract this to an [external code sample](https://www.playframework.com/documentation/2.5.x/Documentation#Code-samples).

File: documentation/manual/working/commonGuide/configuration/ConfigFile.md [hunk 2]
```diff
@@ -19,6 +19,21 @@ At runtime, the default `application.conf` is loaded from the classpath. System
...
+// With Play Scala
+
+import javax.inject._
+import play.api.Configuration
+
+class MyController @Inject() (config: Configuration) {
```
Comment (marcospereira) [inline documentation/manual/working/commonGuide/configuration/ConfigFile.md]: Also, see #6001.

**Comment by cchantep** [comment]
Should be ok

**Comment by wsargent** [comment]
```
[error] Could not find source segment dependency-injection at /home/travis/build/playframework/playframework/documentation/manual/working/commonGuide/configuration/ConfigFile.md:26
[error] @[dependency-injection](code/InjectConfiguration.scala)
[error]   ^
java.lang.RuntimeException: Documentation validation failed
	at com.typesafe.play.docs.sbtplugin.PlayDocsValidation$$anonfun$16.apply(PlayDocsValidation.scala:436)
	at com.typesafe.play.docs.sbtplugin.PlayDocsValidation$$anonfun$16.apply(PlayDocsValidation.scala:312)
	at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
	at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
	at sbt.std.Transform$$anon$4.work(System.scala:63)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
	at sbt.Execute.work(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[error] (Play-Documentation/*:validateDocs) Documentation validation failed
```

**Comment by cchantep** [comment]
fixed

**Pull request merged**